From 7d764c7c2555e8287351961d72be3ebec4d8743f Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Fri, 12 Jul 2024 10:18:53 +0200
Subject: [PATCH] Fix php_pcre_match_impl with 8.4

---
 pcov.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pcov.c b/pcov.c
index f04ba9c..4f7d158 100644
--- a/pcov.c
+++ b/pcov.c
@@ -137,7 +137,11 @@ static zend_always_inline zend_bool php_pcov_wants(zend_string *filename) { /* {
 				ZSTR_VAL(filename), ZSTR_LEN(filename),
 #endif
 				&match, NULL,
+#if PHP_VERSION_ID >= 80400
+				false, 0, 0);
+#else
 				0, 0, 0, 0);
+#endif
 
 			if (zend_is_true(&match)) {
 				zend_hash_add_empty_element(

